### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by DashLt**
I added an exception handler to Worker.run so a single failed task doesn't crash the whole worker, and also added logic to check for HTML responses (error pages that give a 200 response code instead of 404) and zero-length responses (I've run into them, surprisingly).

### CODE REVIEW

File: git-dumper.py [hunk 1]
```diff
@@ -309,13 +332,12 @@ def fetch_git(url, directory, jobs, retry, timeout):
     response = requests.get('%s/.git/HEAD' % url, verify=False, allow_redirects=False)
     printf('[%d]\n', response.status_code)
 
-    if response.status_code != 200:
-        printf('error: %s/.git/HEAD does not exist\n', url, file=sys.stderr)
-        return 1
-    elif not response.text.startswith('ref:'):
-        printf('error: %s/.git/HEAD is not a git HEAD file\n', url, file=sys.stderr)
```
Comment (arthaud) [inline git-dumper.py:316]: Could we keep this check?

File: git-dumper.py [hunk 2]
```diff
@@ -309,13 +332,12 @@ def fetch_git(url, directory, jobs, retry, timeout):
...
+    verification = verify_response(response)
+    if not verification[0]:
+        printf(verification[1], url, "/.git/HEAD")
         return 1
 
+
```
Comment (arthaud) [inline git-dumper.py:340]: nitpick: remove the empty newline

**Review by arthaud — CHANGES_REQUESTED** [review]
That looks pretty good, thanks for working on this. Please address my feedback and I will happily merge this :)

File: git-dumper.py [hunk 3]
```diff
@@ -29,7 +30,7 @@ def printf(fmt, *args, file=sys.stdout):
 
 def is_html(response):
     ''' Return True if the response is a HTML webpage '''
-    return '<html>' in response.text
+    return "html" in response.headers["Content-Type"]
```
[the remark this answers is not in the document]
Comment (DashLt) [reply git-dumper.py:33]: You're right, good catch.

File: git-dumper.py [hunk 4]
```diff
@@ -105,7 +116,12 @@ def run(self):
             if task is None: # end signal
                 return
 
-            result = self.do_task(task, *self.args)
+            try:
+                result = self.do_task(task, *self.args)
+            except Exception as e:
```
[the remark this answers is not in the document]
Comment (DashLt) [reply git-dumper.py:121]: Oh, you're right, good catch

**Comment by arthaud** [comment]
Could you please apply my suggested changes and update the pull request?

**Comment by JHHAX** [comment]
I would love if this could be approved, this is an amazing change

**Comment by arthaud** [comment]
Since the author wouldn't apply my suggested changed, I did that myself. Thanks for the contribution!

**Pull request closed without merging**